tools/scripts/clang_tidy_check.sh

```diff
@@ -65,7 +65,57 @@ if [[ -n "$noSpaceInSuppressions" ]]; then
     exit 1
 fi
 
+function scanWithFileList() {
+    FILE_WITH_SCAN_LIST=$1
+    FILE_TO_SCAN=$2
+
+    if ! test -f "$FILE_WITH_SCAN_LIST"
+    then
+        echo "Scan list file '${FILE_WITH_SCAN_LIST}' does not exist"
+        exit 1
+    fi
+
+    for LINE in $(cat $FILE_WITH_SCAN_LIST); do
```

Comment:
To read lines rather than words, pipe/redirect to a while read loop: https://www.shellcheck.net/wiki/SC2013